ccl/backupccl: TestDataDriven_restore_schema_only failed

ccl/backupccl.TestDataDriven_restore_schema_only [failed](https://tanzanite.cluster.engflow.com/invocations/default/53904df9-b52d-4e8a-991d-36ca628bee33?testReportRun=5&testReportShard=10&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on release-24.1 @ [369e94ddad5fc4d33038a9a6b5303cef1b19df99](https://github.com/cockroachdb/cockroach/commits/369e94ddad5fc4d33038a9a6b5303cef1b19df99):


```
=== RUN   TestDataDriven_restore_schema_only
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestDataDriven_restore_schema_only1245042873
    test_log_scope.go:81: use -show-logs to present logs inline
    datadriven_test.go:446: Mock HTTP Storage "http://127.0.0.1:38779"
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only:3:
        new-cluster [2 args]
        <no input to command>
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only:6:
        exec-sql [0 args]
        CREATE DATABASE d;
        CREATE TYPE d.greeting AS ENUM ('hello', 'howdy', 'hi');
        CREATE TABLE d.t1 (x INT);
        INSERT INTO d.t1 VALUES (1), (2), (3);
        CREATE TABLE d.t2 (x d.greeting);
        INSERT INTO d.t2 VALUES ('hello'), ('howdy');
        COMMENT ON TABLE d.t1 IS 'This comment better get restored from the backed up system table!';
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only:16:
        query-sql [0 args]
        SHOW CREATE TABLE d.t1;
        ----
        d.public.t1 CREATE TABLE public.t1 (
        	x INT8 NULL,
        	rowid INT8 NOT VISIBLE NOT NULL DEFAULT unique_rowid(),
        	CONSTRAINT t1_pkey PRIMARY KEY (rowid ASC)
        );
        COMMENT ON TABLE public.t1 IS 'This comment better get restored from the backed up system table!'
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only:29:
         SELECT id FROM system.namespace WHERE name = 'defaultdb'
        expected:
        100
        
        found:
        102
    testutils.go:289: no Invalid Descriptors
    testutils.go:266: Warning: Could not check for invalid descriptors: sql: database is closed
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestDataDriven_restore_schema_only1245042873
--- FAIL: TestDataDriven_restore_schema_only (1.21s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=5</code>
 - <code>shard=10</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/test-eng
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven_restore_schema_only.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-37364

--- a/pkg/ccl/backupccl/testdata/backup-restore/import-epoch
+++ b/pkg/ccl/backupccl/testdata/backup-restore/import-epoch
@@ -17,12 +17,16 @@ CREATE TABLE baz (i INT PRIMARY KEY, s STRING);
 INSERT INTO baz VALUES (1, 'x'),(2,'y'),(3,'z');
 ----
 
+let $d_id
+SELECT id FROM system.namespace WHERE name = 'foo'
+----
+
 exec-sql
 CREATE VIEW import_epoch (epoch, type)
 AS WITH tbls AS (
    	SELECT id, crdb_internal.pb_to_json('cockroach.sql.sqlbase.Descriptor', descriptor) AS orig FROM system.descriptor
    )
-   SELECT orig->'table'->'importEpoch', orig->'table'->'importType' FROM tbls WHERE id = '109';
+   SELECT orig->'table'->'importEpoch', orig->'table'->'importType' FROM tbls WHERE id = '$d_id';
 ----
 
 exec-sql
@@ -34,10 +38,6 @@ exec-sql
 IMPORT INTO foo (i,s) CSV DATA ('nodelocal://1/export1/export*-n*.0.csv')
 ----
 
-query-sql
-SELECT name, id FROM system.namespace WHERE name = 'foo';
-----
-foo 109
 
 query-sql
 SELECT * FROM import_epoch

--- a/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only
+++ b/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only
@@ -25,11 +25,10 @@ COMMENT ON TABLE public.t1 IS 'This comment better get restored from the backed 
 
 # drop and create defaultDB to ensure it has a higher ID than by default. We will check that when
 # this cluster is restored, the default db with the higher id was also restored
-# by default, default db has an id of 100.
-query-sql
+# by default.
+let $d_id
 SELECT id FROM system.namespace WHERE name = 'defaultdb'
 ----
-100
 
 exec-sql
 DROP DATABASE defaultdb;
@@ -37,7 +36,7 @@ CREATE DATABASE defaultdb;
 ----
 
 query-sql
-SELECT count(*) FROM system.namespace WHERE name = 'defaultdb' AND id > 100
+SELECT count(*) FROM system.namespace WHERE name = 'defaultdb' AND id > $d_id
 ----
 1
 
@@ -110,7 +109,7 @@ COMMENT ON TABLE public.t1 IS 'This comment better get restored from the backed 
 
 # Ensure the defaultdb from the backed up cluster was restored.
 query-sql
-SELECT count(*) FROM system.namespace WHERE name = 'defaultdb' AND id > 100
+SELECT count(*) FROM system.namespace WHERE name = 'defaultdb' AND id > $d_id
 ----
 1
 

